main.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/spf13/viper"
+)
+
+// struct to hold the name field from the JSON request body
+type testRequest struct {
+	Name string `json:"name"`
+}
+
+// struct to hold the new concatination of the user's name
+type testResponse struct {
+	Response string
+}
+
+func main() {
+
+	// Bind the viper key "prt" to the env variable "PORT" from the
+	// dockerfile then get the value from the "PORT" env variable
+	viper.BindEnv("prt", "PORT")
```

Comment:
`port` is a super common flag name, so it might be better to do `viper.BindEnv("port", "PORT")` instead.
you should also checkout golang's init function: https://golang.org/doc/effective_go.html#init